crates/sui-e2e-tests/tests/full_node_tests.rs

```diff
@@ -1369,3 +1370,106 @@ async fn transfer_coin(
     let resp = context.execute_transaction_must_succeed(txn).await;
     Ok((object_to_send.0, sender, receiver, resp.digest, gas_object))
 }
+
+#[sim_test]
+async fn test_full_node_run_with_range_checkpoint() -> Result<(), anyhow::Error> {
+    telemetry_subscribers::init_for_testing();
+    let want_run_with_range = RunWithRange::Checkpoint(2);
+    let test_cluster = TestClusterBuilder::new()
+        .with_epoch_duration_ms(10_000)
+        // This will also do aggressive pruning and compaction of the snapshot
+        .with_enable_db_checkpoints_fullnodes()
+        .with_fullnode_run_with_range(want_run_with_range)
+        .build()
+        .await;
+
+    let _checkpoint_path = test_cluster
+        .fullnode_handle
+        .sui_node
+        .with(|node| node.db_checkpoint_path());
+
+    // wait for node to signal that we reached and processed our desired epoch
+    let got_run_with_range = test_cluster
+        .wait_for_run_with_range_shutdown_signal_with_timeout(Duration::from_secs(120))
+        .await;
+
+    assert_eq!(got_run_with_range, want_run_with_range);
+
+    // we dont want transaction orchestrator enabled when run_with_range != None
+    assert!(test_cluster
+        .fullnode_handle
+        .sui_node
+        .with(|node| node.transaction_orchestrator())
+        .is_none());
+    Ok(())
+}
+
+#[sim_test]
+async fn test_full_node_run_with_range_epoch() -> Result<(), anyhow::Error> {
```

Comment:
this one does not -> `cargo-simtest simtest --test full_node_tests test_full_node_run_with_range_epoch`